packages/deployer/subtasks/prepare-deployment.js

```diff
@@ -25,33 +29,55 @@ subtask(
 ).setAction(async (taskArguments, hre) => {
   const { instance, alias } = taskArguments;
 
-  const deploymentsFolder = path.join(
-    hre.config.deployer.paths.deployments,
-    hre.network.name,
-    instance
-  );
+  const info = {
+    folder: path.resolve(hre.config.paths.root, hre.config.deployer.paths.deployments),
+    network: hre.network.name,
+    instance,
+  };
 
-  // Make sure the deployments folder exists
-  mkdirp.sync(deploymentsFolder);
+  const deploymentsFolder = getDeploymentFolder(info);
+  const deployments = getAllDeploymentFiles(info);
 
-  const { previousFile, currentFile } = await _determineDeploymentFiles(deploymentsFolder, alias);
+  // Make sure the deployments folders exists
+  await mkdirp(deploymentsFolder);
+  await mkdirp(path.join(deploymentsFolder, 'extended'));
 
-  hre.deployer.deployment = autosaveObject(currentFile, DEPLOYMENT_SCHEMA);
-  hre.deployer.deployment.file = relativePath(currentFile);
+  const { currentFile, previousFile } = await _determineDeploymentFiles({
+    deployments,
+    deploymentsFolder,
+    alias,
+  });
+  const { sources, abis } = getDeploymentExtendedFiles(currentFile);
+
+  hre.deployer.paths.deployment = currentFile;
```

Comment:
This shouldn't be possible since `hre.deployer` is semi-frozen?

I like it tho. Perhaps you should see why the freeze is not working, and be declarative in `extensions.js` with this info.

Also, given that you're doing this, wouldn't it be redundant to also store `hre.deployer.deployment.file`? I'd suggest that we remove the latter if we use the paths defined above.

Reply:
Yup, `hre.deployer.deployment.file` is not there anymore. And the prevent extension is not working because is not deep on its effect. So, the `hre.deployer` object is blocked, but `hre.deployer.paths` not.

Reply:
Done, fixed it just adding `Object.preventExtensions(hre.deployer.paths);`  :)

Reply:
Awesome